flocker/node/_docker.py

```diff
@@ -508,6 +508,9 @@ def _remove():
                     break
 
             try:
+                while (self._client.inspect_container(container_name)
+                        ['State']['Running']):
+                    sleep(0.1)
```

Comment:
I don't see this as a bug in Docker. It is pretty clear that the stop operation puts the container into a stopping state and then returns to the client before the container is actually stopped.

As for additional failures, I think Monday was a bad day to test. All branches were giving additional errors, mainly to do with database and move failures in the acceptance tests.  That's the problem with having multiple intermittent errors. It can be difficult to isolate them.  However, tests before Monday were working fine for this branch. See http://build.clusterhq.com/builders/flocker%2Finstalled-package%2Ffedora-20/builds/1637 

Dealing with Docker restarts is a separate issue. Although I would expect a Docker stop operation to override any restart policy. Do you have example code to demonstrate this problem? 

I have added further logging to help debugging any future problems, e.g.  [FLOC-1729]